dspace-server-webapp/src/main/java/org/dspace/app/rest/authorization/impl/CanChangePasswordFeature.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * The contents of this file are subject to the license and copyright
+ * detailed in the LICENSE and NOTICE files at the root of the source
+ * tree and available online at
+ *
+ * http://www.dspace.org/license/
+ */
+package org.dspace.app.rest.authorization.impl;
+
+import java.sql.SQLException;
+
+import org.apache.commons.lang.StringUtils;
+import org.dspace.app.rest.authorization.AuthorizationFeature;
+import org.dspace.app.rest.authorization.AuthorizationFeatureDocumentation;
+import org.dspace.app.rest.model.BaseObjectRest;
+import org.dspace.app.rest.model.EPersonRest;
+import org.dspace.core.Context;
+import org.springframework.stereotype.Component;
+
+/**
+ * The canChangePassword feature. It can be used to verify if the user can change his password.
+ */
+@Component
+@AuthorizationFeatureDocumentation(name = CanChangePasswordFeature.NAME,
+        description = "It can be used to verify if the user can change his password")
+public class CanChangePasswordFeature implements AuthorizationFeature {
+
+    public static final String NAME = "canChangePassword";
+
+    @Override
+    public boolean isAuthorized(Context context, BaseObjectRest object) throws SQLException {
+        if (context.getCurrentUser() != null && StringUtils.equals(context.getAuthenticationMethod(), "password")) {
```

Comment:
@benbosman does it makes sense to have this configurable? Meaning, have a property to enable/disable this?

Reply:
@paulo-graca I don't think is was possible in DSpace 6, but making it configurable would only be a very minor change (most of the work goes to ITs to test both options)
I will wait for the Angular work to be completed to make sure there's enough budget left